File: nysol_python/mcmd/methods/convTex/db/smp/func_regexstr.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='regexstr'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,str
1,xcbbbayy
2,xxcbaay
3,
4,bacabbca
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
正規表現 ``c.*a`` に最も長くマッチする部分文字列を得る。
``id=2`` では、 ``cba`` もしくは ``cbaa`` いずれの部分文字列にもマッチしたと考えることができるが、
本関数では、より長くマッチした文字列を返す。
'''
script['sh_code']='''
mcal c='regexstr($s{str},"c.*a")' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='regexstr($s{str},"c.*a")', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_regexstr.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,str
1,xcbbbayy
2,xxcbaay
3,
4,bacabbca
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Extract the longest substring that matches the regular expression \verb|c.*a|. At row where \verb|id=2|, the regular expression matches the string \verb|cba| and \verb|cbaa|, however, the longer substring is returned.
EOF
scp=<<'EOF'
more dat1.csv
mcal c='regexstr($s{str},"c.*a")' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

